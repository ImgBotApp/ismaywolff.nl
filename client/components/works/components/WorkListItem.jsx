import React, { PropTypes } from 'react'
import { Link } from 'react-router-dom'
import styled from 'styled-components'
import { ThumbContainer } from '../../images'

const StyledLink = styled(Link)`
  border: 1px dotted black;
  display: block;
  padding: 3px;
`

function WorkListItem({ work }) {
  return (
    <div>
      <StyledLink to={`/work/${work.slug}`}>
        <ThumbContainer id={work.thumbnail} />
      </StyledLink>
    </div>
  )
}

WorkListItem.propTypes = {
  work: PropTypes.shape({
    description: PropTypes.string.isRequired,
    slug: PropTypes.string.isRequired,
    published: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    type: PropTypes.string.isRequired
  }).isRequired
}

export default WorkListItem
