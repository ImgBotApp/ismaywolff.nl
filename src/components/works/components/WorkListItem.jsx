import React, { PropTypes } from 'react'

function WorkListItem({ work }) {
  return <div>{work && work.title}</div>
}

WorkListItem.propTypes = {
  work: PropTypes.object.isRequired
}

export default WorkListItem
