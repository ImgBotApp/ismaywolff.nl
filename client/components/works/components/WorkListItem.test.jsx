import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import WorkListItem from './WorkListItem'

describe('<WorkListItem />', () => {
  it('renders correctly', () => {
    const mockWork = {
      description: 'description',
      published: 'published',
      slug: 'title',
      title: 'title',
      type: 'type'
    }
    const wrapper = shallow(<WorkListItem work={mockWork} />)
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })
})
