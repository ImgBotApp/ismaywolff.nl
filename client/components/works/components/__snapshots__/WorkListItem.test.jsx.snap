exports[`<WorkListItem /> renders correctly 1`] = `
<div>
  <Styled(Link)
    to="/work/title">
    <Connect(ThumbContainer) />
  </Styled(Link)>
</div>
`;
